File: M1/C/gcd.c
#include <stdio.h>

int gcd(int a, int b);

int gcd(int a, int b) {
    int x = a;
    int y = b;

    while (x != y) {
        if (x > y) {
           x = x - y;
        } else {
           y = y - x;
        }
    }

    return x;
}

int main(void) {
    int i1 = 42;
    int i2 = 1024;

    printf("The GCD of %d and %d is %d\n",
           i1, i2, gcd(i1, i2));

    return 0;
}

File: M1/Rust/src/bin/gcd.rs
use std::io;

fn gcd(a: i32, b: i32) -> i32 {
    let mut x = a;
    let mut y = b;

    while x != y {
        if x > y {
            x = x - y;
        } else {
            y = y - x;
        }
    }

    return x;
}

fn main() {
    let i1: i32 = 42;
    let i2: i32 = 1024;

    println!("GCD of {} and {} is {}",
             i1, i2, gcd(i1, i2));
}
